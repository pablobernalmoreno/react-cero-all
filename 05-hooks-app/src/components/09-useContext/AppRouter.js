import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <Switch>
          <Route path="/about" component={AboutScreen} />
          <Route path="/login" component={LoginScreen} />
          <Route path="/" component={HomeScreen} />
          <Route component={HomeScreen} />
        </Switch>
      </div>
    </Router>
  );
};
