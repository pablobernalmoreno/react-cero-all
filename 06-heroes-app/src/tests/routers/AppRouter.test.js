import { mount } from "enzyme";
import { AuthContext } from "../../auth/AuthContext";
import { AppRouter } from "../../routers/AppRouter";

describe("Pruebas en <AppRoute />", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };

  test("debe de mostrar login si no está autenticado", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar el componente '<DcScreen /> si está autenticado", () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged: true,
        name: "Pablo",
      },
    };
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper.find(".navbar").exists()).toBeTruthy();
  });
});
