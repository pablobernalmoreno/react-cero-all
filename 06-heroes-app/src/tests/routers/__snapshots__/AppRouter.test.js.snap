// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRoute /> debe de mostrar login si no est√° autenticado 1`] = `
<AppRouter>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "REPLACE",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "41eszl",
            "pathname": "/login",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <Switch>
          <PublicRoute
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/login",
                "url": "/login",
              }
            }
            isAuthenticated={false}
            location={
              Object {
                "hash": "",
                "key": "41eszl",
                "pathname": "/login",
                "search": "",
                "state": undefined,
              }
            }
            path="/login"
          >
            <Route
              component={[Function]}
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/login",
                  "url": "/login",
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "41eszl",
                  "pathname": "/login",
                  "search": "",
                  "state": undefined,
                }
              }
              path="/login"
            >
              <component
                history={
                  Object {
                    "action": "REPLACE",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "41eszl",
                      "pathname": "/login",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "key": "41eszl",
                    "pathname": "/login",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/login",
                    "url": "/login",
                  }
                }
              >
                <LoginScreen
                  history={
                    Object {
                      "action": "REPLACE",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "key": "41eszl",
                        "pathname": "/login",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "key": "41eszl",
                      "pathname": "/login",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/login",
                      "url": "/login",
                    }
                  }
                >
                  <div
                    className="container mt-5"
                  >
                    <h1>
                      Login Screen
                    </h1>
                    <hr />
                    <button
                      className="btn btn-primary"
                      onClick={[Function]}
                    >
                      Login
                    </button>
                  </div>
                </LoginScreen>
              </component>
            </Route>
          </PublicRoute>
        </Switch>
      </div>
    </Router>
  </BrowserRouter>
</AppRouter>
`;
